bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -402,7 +402,12 @@ class PlotCanvasView extends Renderer.View
     # reach the frame in time (sometimes) so force an update here for now
     @model.get('frame')._update_mappers()
 
+    # Get context and restore to its initial state. The restoring
+    # Would stricly speaking not be necessary, but in case there is an error
+    # during drawing, restore() won't be called enough. This guards for that.
     ctx = @canvas_view.ctx
+    for i in [0..3]
+      ctx.restore()
```

Comment:
AFAIK JS does support `finally`: http://www.w3schools.com/jsref/jsref_try_catch.asp So that would be my suggestion to try